src/mouse_events.cpp

```diff
@@ -604,11 +604,22 @@ void mouse_handler::mouse_button_event(const SDL_MouseButtonEvent& event, uint8_
 {
 	if (loc == map_location::null_location())
 		return;
+	if (button < SDL_BUTTON_LEFT || button > SDL_BUTTON_X2)
+		return;
+
+	static const std::string buttons[6] = {
+		[0]					= "",
+		[SDL_BUTTON_LEFT]	= "left",
+		[SDL_BUTTON_MIDDLE]	= "middle",
+		[SDL_BUTTON_RIGHT]	= "right",
+		[SDL_BUTTON_X1]		= "x1",
+		[SDL_BUTTON_X2]		= "x2",
+	};
```

Comment:
Okay, so I have two comments here.

1. Unfortunately, designated initializers are new in C++20, but Wesnoth builds as C++17. There are a couple of options here; I don't think compatibility wrappers can work for this feature, but I'm willing to be proven wrong; or you could use a mapping type (`std::map` or `std::unordered_map`).
2. `x1` and `x2` seem a bit too obtuse for my liking. I think I'd prefer using `ext1` and `ext2`, or maybe `extra` or `extended`.

Reply:
This has still not been addressed.

Reply:
Sorry I missed this! Hrm, I could always just use a plain array initializer and put the enums in comments. I like having the code prove that the mapping is correct. You can probably tell that I've done most of my code in C for the past few years. :)

```C++
static const std::string buttons[6] = {
	"",
	"left",		// SDL_BUTTON_LEFT
	"middle",	// SDL_BUTTON_MIDDLE
	"right",	// SDL_BUTTON_RIGHT
	"x1",		// SDL_BUTTON_X1
	"x2",		// SDL_BUTTON_X2
};
```
?

Reply:
I definitely like the idea of using designated initializers here too, but I'm pretty certain it's going to fail to compile on some platform we support. (It would work if we enabled GNU extensions, but then that could cause problems on Windows.)

That proposal isn't how I'd do it, but I don't really have a problem with it.

Don't forget item 2 of my previous comment.

Reply:
Thanks, got the grammar too!

Reply:
"mouse-4" and "mouse-5"?